src/components/cc-ssh-key-list/cc-ssh-key-list.js

```diff
@@ -0,0 +1,438 @@
+              (${this.keyData.githubKeys.length})
+            ` : ''}
+          </div>
+          <div slot="info">${i18n('cc-ssh-key-list.github.info')}</div>
+
+          ${state === 'loading' ? html`
+            <div class="key-list">${this._renderKeyList('skeleton')}</div>
+          ` : ''}
+
+          ${state === 'loaded' && !this.keyData.isGithubLinked ? html`
+            <span class="info-msg">${i18n('cc-ssh-key-list.github.unlinked')}</span>
+          ` : ''}
+
+          ${state === 'loaded' && this.keyData.isGithubLinked ? html`
+            ${this.keyData.githubKeys.length === 0 ? html`
+              <span class="info-msg">${i18n('cc-ssh-key-list.github.empty')}</span>
+            ` : ''}
+            ${this._renderKeyList('github', this.keyData.githubKeys)}
+          ` : ''}
+
+          ${state === 'error' ? html`
+            <span class="info-msg">${i18n('cc-ssh-key-list.error.loading')}</span>
+          ` : ''}
+        </cc-block-section>
+
+        <!-- documentation link -->
+        <cc-block-section>
+          <div class="align-end">
+            ${i18n('cc-ssh-key-list.doc.info')}
+          </div>
+        </cc-block-section>
+      </cc-block>
+    `;
+  }
+
+  _renderCreateSshKeyForm () {
+    return html`
+      <div class="create-form">
+        <cc-input-text
+          ?disabled=${this.createSshKeyForm.state === 'creating'}
+          @cc-input-text:requestimplicitsubmit=${this._onCreateKey}
+          .value="${live(this.createSshKeyForm.name?.value)}"
+          class="create-form__name"
+          label=${i18n('cc-ssh-key-list.add.name')}
+          required
+          ${ref(this._createFormNameRef)}
+        >
+          ${this.createSshKeyForm.name.error === 'required' ? html`
+            <p slot="error">${i18n('cc-ssh-key-list.error.empty.name')}</p>
+          ` : ''}
+        </cc-input-text>
+        <cc-input-text
+          ?disabled=${this.createSshKeyForm.state === 'creating'}
+          @cc-input-text:requestimplicitsubmit=${this._onCreateKey}
+          .value="${live(this.createSshKeyForm.publicKey?.value)}"
+          class="create-form__public-key"
+          label=${i18n('cc-ssh-key-list.add.public-key')}
+          required
+          ${ref(this._createFormPublicKeyRef)}
+        >
+          ${this.createSshKeyForm.publicKey.error === 'required' ? html`
+            <p slot="error">${i18n('cc-ssh-key-list.error.empty.public-key')}</p>
+          ` : ''}
+        </cc-input-text>
+        <div class="create-form__footer">
+          <cc-button
+            @cc-button:click=${this._onCreateKey}
+            class="create-form__add-btn"
+            primary
+            ?waiting=${this.createSshKeyForm.state === 'creating'}
+          >
+            ${i18n('cc-ssh-key-list.add.btn')}
+          </cc-button>
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * @param {"personal"|"github"|"skeleton"} type
+   * @param {SshKey[]} keys
+   */
+  _renderKeyList (type, keys = [{}]) {
```

Comment:
I'm not very comfortable with this single entry that is used for skeleton type.
I would prefer to be more explicit:

```javascript
_renderKeyList (type, keys = []) {
  const skeleton = (type === 'skeleton');

  if (skeleton) {
    keys = [
      {
        state: 'idle',
        name: fakeString(32),
        fingerprint: fakeString(32),
      },
    ];
  }
  // ...
}
```

Reply:
We were also a bit "is it really OK?" but I also don't like when we modify the reference of a function param. We can find something better together ;-)